aptos-move/block-executor/src/executor.rs

```diff
@@ -910,6 +911,19 @@ where
         };
 
         loop {
+            if let SchedulerTask::ValidationTask(_, incarnation, _) = &scheduler_task {
+                if *incarnation as usize > num_txns + 10 {
```

Comment:
The main scenario where I want this fallback to save us is some sort of livelock where things keep getting invalidated and re-executed, maybe preventing some worker from resolving the state. I've managed to get into such a state while playing with some faulty executor logic. In that scenario, having a large number is good enough, but it is true that the number should depend on number of workers. I suggest num_workers ^ 2 term - my thinking is it doesn't need to be theoretically impossible: it should be unlikely enough that when it happens 
(a) it's worth investigating - potentially changing the threshold or doing different optimization (such as delaying the txn when it's re-executing too much as you suggested), and 
(b) it's already slow enough that sequential fallback wouldn't do much harm.

the reason worker^2 seems reasonable is because esp. with backup, rolling commit should reach the current transaction in about num_workers many executions (with dependency waiting it's harder to model, but waiting also avoids re-executions). We can be conservative and even have num_txns + num_workers ^2, or via some constants. what do you think?

Reply:
this PR is good to go after you address this comment

Reply:
done